src/pages/staticPages/tools/oncoprinter/Oncoprinter.tsx

```diff
@@ -0,0 +1,338 @@
+import * as React from "react";
+import {observer, Observer} from "mobx-react";
+import {action, computed, IObservableObject, observable} from "mobx";
+import Oncoprint from "../../../../shared/components/oncoprint/Oncoprint";
+import OncoprintControls, {
+    IOncoprintControlsHandlers,
+    IOncoprintControlsState
+} from "shared/components/oncoprint/controls/OncoprintControls";
+import {Sample} from "../../../../shared/api/generated/CBioPortalAPI";
+import {percentAltered} from "../../../../shared/components/oncoprint/OncoprintUtils";
+import AppConfig from "appConfig";
+import OncoprintJS from "oncoprintjs";
+import fileDownload from "react-file-download";
+import svgToPdfDownload from "shared/lib/svgToPdfDownload";
+import classNames from "classnames";
+import FadeInteraction from "shared/components/fadeInteraction/FadeInteraction";
+import OncoprinterStore from "./OncoprinterStore";
+import autobind from "autobind-decorator";
+
+interface IOncoprinterProps {
+    divId: string;
+    store: OncoprinterStore;
+}
+
+@observer
+export default class Oncoprinter extends React.Component<IOncoprinterProps, {}> {
+
+    @observable distinguishMutationType:boolean = true;
+    @observable sortByMutationType:boolean = true;
+    @observable sortByDrivers:boolean = true;
+
+    @observable showUnalteredColumns:boolean = true;
+    @observable showWhitespaceBetweenColumns:boolean = true;
+
+    @observable showMinimap:boolean = false;
+
+    @observable horzZoom:number = 0.5;
+
+    @observable mouseInsideBounds:boolean = false;
+
+    @observable renderingComplete = true;
+
+    private controlsHandlers:IOncoprintControlsHandlers;
+    private controlsState:IOncoprintControlsState & IObservableObject;
+
+    @observable.ref private oncoprint:OncoprintJS<any>;
+
+    constructor(props:IOncoprinterProps) {
+        super(props);
+
+        (window as any).oncoprinter = this;
+
+        const self = this;
+
+        this.controlsHandlers = this.buildControlsHandlers();
+
+        this.controlsState = observable({
+            get showUnalteredColumns() {
```

Comment:
this are automatically made computed by calling observable on parent?

Reply:
yeah https://mobx.js.org/refguide/object.html 
`Property getters will be automatically turned into derived properties, just like @computed would do.`